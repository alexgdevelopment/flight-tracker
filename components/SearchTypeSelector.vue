<template>
  <v-radio-group v-model="searchType" row>
    <v-spacer />
    <v-radio color="orange" :value="searchTypeVar.AIRPORT">
      <template #label>
        <div class="orange--text">Airports</div>
      </template>
    </v-radio>
    <v-spacer />
    <v-radio color="blue" :value="searchTypeVar.FLIGHT">
      <template #label>
        <div class="blue--text">Flights</div>
      </template>
    </v-radio>
    <v-spacer />
  </v-radio-group>
</template>
<script>
  import searchTypeMixin from '@/assets/mixins/searchTypeMixin.js'
  export default {
    mixins: [searchTypeMixin],
    computed: {
      searchType: {
        get() {
          return this.$store.state.searchType
        },
        set(value) {
          this.$store.dispatch('changeSearchType', { searchType: value })
        },
      },
    },
  }
</script>