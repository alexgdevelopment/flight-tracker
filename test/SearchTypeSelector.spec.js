import { fireEvent } from '@testing-library/dom'
import { Store } from 'vuex'
import { renderWithVuetify } from './helpers'
import SearchTypeSelector from '@/components/SearchTypeSelector.vue'

describe('Search type selector', () => {
  let wrapper = null
  let store
  let state
  let actions

  beforeEach(() => {
    state = {
      mainColor: 'green',
    }
    actions = {
      changeSearchType: jest.fn(),
    }
    store = new Store({
      state,
      actions,
    })
    wrapper = renderWithVuetify(SearchTypeSelector, {
      store,
    })
  })

  test('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should display "Airports" and "Flights" radio buttons', () => {
    expect(wrapper.getByText('Airports'))
    expect(wrapper.getByText('Flights'))
  })

  test('should emit the proper event on button select', () => {
    fireEvent.click(wrapper.getByText('Airports'))
    expect(actions.changeSearchType).toHaveBeenCalled()
  })
})
