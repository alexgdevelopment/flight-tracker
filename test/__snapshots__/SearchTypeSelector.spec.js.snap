// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search type selector should match the snapshot 1`] = `
"<div class=\\"v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--row\\">
  <div class=\\"v-input__control\\">
    <div class=\\"v-input__slot\\" style=\\"height: auto;\\">
      <div role=\\"radiogroup\\" aria-labelledby=\\"input-4\\" class=\\"v-input--radio-group__input\\">
        <div class=\\"spacer\\"></div>
        <div class=\\"v-radio theme--light\\">
          <div class=\\"v-input--selection-controls__input\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-radiobox-blank theme--light\\"></i><input aria-checked=\\"false\\" id=\\"input-5\\" role=\\"radio\\" type=\\"radio\\" name=\\"radio-4\\" value=\\"search_type_airport\\">
            <div class=\\"v-input--selection-controls__ripple\\"></div>
          </div><label for=\\"input-5\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: relative;\\">
            <div class=\\"orange--text\\">Airports</div>
          </label>
        </div>
        <div class=\\"spacer\\"></div>
        <div class=\\"v-radio theme--light\\">
          <div class=\\"v-input--selection-controls__input\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-radiobox-blank theme--light\\"></i><input aria-checked=\\"false\\" id=\\"input-7\\" role=\\"radio\\" type=\\"radio\\" name=\\"radio-4\\" value=\\"search_type_flight\\">
            <div class=\\"v-input--selection-controls__ripple\\"></div>
          </div><label for=\\"input-7\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: relative;\\">
            <div class=\\"blue--text\\">Flights</div>
          </label>
        </div>
        <div class=\\"spacer\\"></div>
      </div>
    </div>
    <div class=\\"v-messages theme--light\\">
      <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
    </div>
  </div>
</div>"
`;
